### PR TITLE
feat(protocol-designer): implement thermocyclerProfileStep CC

### LINKED ISSUE #5835: PD: ThermocyclerProfileStep command creator
## Blocked

Needs formToArgs and executeProfile mini CC (see below, I'll make tickets for both soon)

## Overview

This is the command creator that actuates the TC Profile step form.

A new command creator `ThermocyclerProfileStep`...

* It takes `ThermocyclerProfileStepArgs` created in (TODO thermocyclerFormToArgs for profile step, make ticket and design Args data type)

* It uses the "mini command creators" from #5598 (TODO need to make executeProfile mini CC?)

* For the dynamic profile fields: it unpacks the cycles from `ThermocyclerProfileStepArgs` into a flat list of profile steps, taking into account the `repetitions` of each cycle. This flat list is transformed into the `params` of a `'thermocycler/runProfile'` command (ie, to `profile: Array<{| temperature: number, holdTime: number |}>`).

* For "Ending hold" fields: this works similarly to the existing TC State step form. It pulls `ThermocyclerModuleState` out of the previous RobotState and compares that prev state with the new state from `ThermocyclerProfileStepArgs`. Using this diff, it composes the commands needed to transform the previous state to the requested next state. For example, if the lid is closed in the Args and there are no other changes, it will only return [closeLid({moduleId})] is its list of commands. OR MAYBE IT JUST SETS WITHOUT A CARE FOR THE DIFF?

### PR DESCRIPTION
**Pull request by IanLondon**
## overview

Closes #5835

## review requests

- In addition to code review, sanity check this implementation (look at the tests, or save the JSON and look at the `commands`) and make sure the TC Profile step works as you'd expect (aside from runProfile command not yet updating RobotState, that's a separate ticket)

## risk assessment

Low, PD only and under FF

### CODE REVIEW

File: protocol-designer/src/step-generation/commandCreators/compound/thermocyclerProfileStep.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  const thermocyclerState = thermocyclerStateGetter(prevRobotState, moduleId)
+
+  if (thermocyclerState === null) {
+    return { errors: [errorCreators.missingModuleError()] }
+  }
+
+  const commandCreators: Array<CurriedCommandCreator> = []
+
+  if (
+    profileTargetLidTemp !== null && // TODO IMMEDIATELY should `profileTargetLidTemp` be number (not `number | null`)? Check designs: form validation
```
Comment (shlokamin) [inline protocol-designer/src/step-generation/commandCreators/compound/thermocyclerProfileStep.js:37]: has this been resolved?

File: protocol-designer/src/step-generation/__tests__/thermocyclerProfileStep.test.js [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+import type { Command } from '@opentrons/shared-data/protocol/flowTypes/schemaV4'
+import type { ThermocyclerModuleState } from '../../step-forms/types'
+import type { ThermocyclerProfileStepArgs } from '../types'
+
+const temperatureModuleId = 'temperatureModuleId'
+const thermocyclerId = 'thermocyclerId'
+
+describe('thermocyclerProfileStep', () => {
+  // TODO: case with bad module ID should error
+
```
Comment (shlokamin) [inline protocol-designer/src/step-generation/__tests__/thermocyclerProfileStep.test.js:17]: forgot to do this it looks like

**Review by shlokamin — APPROVED** [review]
Looks good! Looks like there's a few forgotten `TODOs`

I think it would also be worth adding a test case when `profileSteps` isn't empty (even though that atomic command creator is unit tested)

**Review by Kadee80 — APPROVED** [review]
🥥 🦀

**Pull request merged**